src/chess/ChessMatch.java

```diff
@@ -155,24 +165,42 @@ private boolean testCheck(Color color) {
         return false;
     }
 
+    private boolean testCheckMate(Color color) {
```

Comment:
Esse método tá muito longo e poderia ser quebrado em outros. 
Quem sabe um para cada for!

só uma idéia!